src/server/auth/server/auth.go

```diff
@@ -47,59 +42,33 @@ const (
 
 	defaultTokenTTLSecs = 14 * 24 * 60 * 60 // two weeks
 
-	publicKey = `-----BEGIN PUBLIC KEY-----
-MIICIjANBgkqhkiG9w0BAQEFAAOCAg8AMIICCgKCAgEAoaPoEfv5RcVUbCuWNnOB
-WtLHzcyQSe4SbtGGQom/X27iq/7s8dcebSsCd2cwYoyKihEQ5OlaghrhcxTTV5AN
-39O6S0YnWjt/+4PWQQP3NpcEhqWj8RLPJtYq+JNrqlyjxBlca7vDcFSTa6iCqXay
-iVD2OyTbWrD6KZ/YTSmSY8mY2qdYvHyp3Ue5ueH3rSkKRUjo4Jyjf59PntZD884P
-yb9kC+weh/1KlbDQ4aV0U9p6DSBkW7dinOQj7a1/ikDoA9Nebnrkb1FF9Hr2+utO
-We4e4yOViDzAP9hhQiBhOVR0F6wJF5i+NfuLit4tk5ViboogEZqIyuakTD6abSFg
-UPqBTDDG0UsVqjnU5ysJ1DKQqALnOrxEKZoVXtH80/m7kgmeY3VDHCFt+WCSdaSq
-1w8SoIpJAZPJpKlDjMxe+NqsX2qUODQ2KNkqfEqFtyUNZzfS9o9pEg/KJzDuDclM
-oMQr1BG8vc3msX4UiGQPkohznwlCSGWf62IkSS6P8hQRCBKGRS5yGjmT3J+/chZw
-Je46y8zNLV7t2pOL6UemdmDjTaMCt0YBc1FmG2eUipAWcHJWEHgQm2Yz6QjtBgvt
-jFqnYeiDwdxU7CQD3oF9H+uVHqz8Jmmf9BxY9PhlMSUGPUsTpZ717ysL0UrBhQhW
-xYp8vpeQ3by9WxPBE/WrxN8CAwEAAQ==
------END PUBLIC KEY-----
-`
-
 	// magicUser is a special, unrevokable cluster administrator. It's not
 	// possible to log in as magicUser, but pipelines with no owner are run as
 	// magicUser when auth is activated.
 	magicUser = `GZD4jKDGcirJyWQt6HtK4hhRD6faOofP1mng34xNZsI`
 )
 
-type authState uint8
-
-const (
-	// If auth is disabled, authentication and authorization are disabled, and any
-	// RPCs sent to the cluster are executed.
-	authDisabled authState = iota
-
-	// If auth is expired, only admins can log in or perform any actions in the
-	// cluster. If an admin calls auth.Activate(), then the cluster will
-	// transition to the 'authEnabled' state. If all cluster admins are removed,
-	// the cluster will transition to the 'authDisabled' state.
-	authExpired
-
-	// If auth is enabled, users can log in, and users can manage access to repos
-	// repos with access control lists
-	authEnabled
-)
-
 // epsilon is small, nonempty protobuf to use as an etcd value (the etcd client
 // library can't distinguish between empty values and missing values, even
 // though empty values are still stored in etcd)
 var epsilon = &types.BoolValue{Value: true}
 
+// authEnterpriseClient contains address and connection info needed to check
+// the cluster's enterprise status (auth is an enterprise feature, and cannot
+// be used if the cluster doesn't have enterprise features enabled)
+type enterpriseClientInfo struct {
```

Comment:
Why is called an `enterpriseClientInfo` instead of just an `enterpriseClient`? The `Info` part doesn't seem to really match with what `Info` normally means after a noun. Also the comment doesn't refer to the actual type name, which I think the linter will complain about.

Reply:
I think I picked enterpriseClientInfo because `client` is already a field, and I thought `client.client` might be weird. But it obviously doesn't matter—fixed.

Reply:
I see, makes sense.